File: 3637.trionic-array-i/solution.kt
class Solution {
    fun isTrionic(nums: IntArray): Boolean {
        val pairs = nums.asList().zipWithNext()
        val left = pairs.dropWhile { (a, b) -> a > b }
        val middle = left.dropWhile { (a, b) -> a < b }
        val right = middle.dropWhile { (a, b) -> a > b }
        val rest = right.dropWhile { (a, b) -> a < b }
        val n1 = nums.size - 1
        return left.size == n1 && middle.size < n1 && right.size < middle.size && right.size > 0 && rest.size == 0
    }
}

File: 3637.trionic-array-i/solution.go
func isTrionic(nums []int) bool {
    i := 0
    for i + 1 < len(nums) && nums[i] < nums[i + 1] {
        i++
    }
    j := i
    for j + 1 < len(nums) && nums[j] > nums[j + 1] {
        j++
    }
    k := j
    for k + 1 < len(nums) && nums[k] < nums[k + 1] {
        k++
    }
    return i > 0 && k == len(nums) - 1 && j - 1 > 0 && k - j > 0
}